CHANGELOG.md

```diff
@@ -56,73 +53,45 @@ class Meta extends Component {
+- **Added**: `beforerender` lifecycle hook.  Its similar to `load` but runs before mounting.
+- **Added**: `afterupdate` runs after `update` returns true and the results of `createElement` is mutated over the mounted component.  Useful for adjusting scroll position.
 - **Fixed**: More robust unmount and remounting behavior.
 
 #### `nanocomponent@5` to `nanocomponent@6` upgrade guide:
 
 - Read through the new leaflet example to get an idea of the differences between the old and new API. 🗺
-- Move any DOM mutation code from `_render` into `_update`.
-- Ensure `_render` returns a DOM node always. (You will get warnings if you don't and it probably won't work)
-- Consider moving any `_load` actions into `_willRender` if they don't depend on the newly rendered node being mounted in a DOM tree yet.
-- Take advantage of `_didUpdate` allowing you to interact with your component after `_render` is called on mounted components 🙌
-
-## 5.2.0
-* Added more lifecycle hooks: `_willMount`, `_didMount`, `_willUpdate` in addition to `_didUpdate`.
-
-## 5.1.0
-* Update [nanomorph](http://ghub.io/nanomorph) to `^5.1.2`.  This adds the new child-reordering algorithm so we get a minor version bump.  Keep an eye out for weird ness and report broken corner cases 🙏
-
-## 5.0.1
-* Fix proxy leak by resetting proxy node ID after DOM removal is detected.
-
-## 5.0.0
-* Update [bel](http://ghub.io/bel) to ^5.0.0
-
-## 5.0.0-1 - 2017-05-16
-* Beta release!  Please let me know if there is anything wrong with this!
-* **Breaking**: Remove `on-load` and use a new DOM ID based DOM tracking system.  Requires ES5 support for getters.
-* **Breaking**: Remove `_load` and `_unload` methods.  You have to wrap instances of `cache-component` with `on-load` on your own now.
-* **Added**: Add `_didUpdate` hook so you can call DOM methods after the component updates.  Handy for updating a scroll position.
-
-## 4.0.2 - 2017-05-05
-* Run _unload before we clear internal references, allowing you to clean up event listeners on `this._element` and anything else you want to do.
-
-## 4.0.1 - 2017-04-10
-* Fix instance clobbering bug.  This bug showed up when you had two instances of the same component morphing over each other.  This would cause the real DOM reference to get lost between the internal _element references of the two instances.  The work around was the introduction of ccId which is a unique ID to prevent this.
-
-## 4.0.0 - 2017-04-10
-* use [on-load](https://github.com/shama/on-load) to invalidate `this._element`.  Fixes component rendering when they get completely removed from the DOM.
-* added `_load` and `_unload` methods to allow you to run code when the DOM is mounted and unmounted.
-* handle morphing internally, and ALWAYSE return a proxy node.  There is no other way.
-
-## 3.0.0 - 2017-04-10
-* initial release
```

Comment:
Yeah it was cache-component history which will live on over at https://github.com/hypermodules/cache-component/blob/master/CHANGELOG.md 

I dont think nanocomponent had a changelog before 6.0.0 but I should double check there wasn't one at some point, and port it over to the new one.